### PR TITLE
Support for github folders

### PR DESCRIPTION
**Pull request by Marsup**
Hey, it's me again :)

I've needed to grab complete folders from github dependencies, so here's a quick patch for that.
Sometimes the GH API rejects for its rate limit, I've added a GITHUB_AUTH_TOKEN as en environment variable for that.

### CODE REVIEW

**Comment by Marsup** [comment]
@dtao thoughts ?

**Comment by dtao** [comment]
Sounds awesome! I'll take a look tonight.

File: bin/deft [hunk 1]
```diff
@@ -55,8 +75,8 @@ function getDestinationPath(fileName) {
   return path.join(process.cwd(), folder, fileName);
 }
 
-function pathExists(path) {
-  return (fs.existsSync || path.existsSync)(path);
+function pathExists(p) {
```
Comment (dtao) [inline bin/deft:78]: Ugh! Can't believe this bug is still there! I actually think I fixed this locally on an old laptop and then never pushed... not that I have proof, mind you...

File: bin/deft [hunk 2]
```diff
@@ -67,16 +87,23 @@ function shouldOverwrite(filePath) {
   return commander.force;
 }
 
-function ensureDirectoryExists(path) {
-  mkdirp.sync(path.dirname(path));
+function ensureDirectoryExists(p) {
```
Comment (dtao) [inline bin/deft:90]: Yeah, same damn embarrassing bug here... thanks for fixing ;)

File: deft.js [hunk 3]
```diff
@@ -115,29 +115,17 @@ deft.normalizeFilePaths = function(files) {
  * @return {string}
  *
  * @examples
- * deft.getUrl('lodash.js', ['lodash/lodash', 'lodash.js']);
- * // => 'https://raw.githubusercontent.com/lodash/lodash/master/lodash.js'
+ * deft.getInformationUrl('lodash.js', ['lodash/lodash', 'lodash.js']);
+ * // => 'https://api.github.com/repos/lodash/lodash/contents/lodash.js?ref=master'
  *
- * deft.getUrl('underscore.js', ['jashkenas/underscore', '1.6.0', 'underscore.js']);
```
Comment (dtao) [inline deft.js:121]: This doesn't look OK to me. Not sure if you're aware, but these examples are actual test cases using [autodoc](https://github.com/dtao/autodoc). It looks like your changes here break existing behavior, which is to support dependencies specified at arbitrary (non-GitHub) URLs. Or am I missing something?

Comment (Marsup) [inline deft.js:121]: Well my deft.json test file contains a download for http://code.angularjs.org/1.2.5/angular-1.2.5.zip and it still works, this function wasn't used for anything else than gh downloads.

Comment (dtao) [inline deft.js:121]: That's not true :( The logic for downloading and extracting archives is separate from downloading files. I just tried with http://code.jquery.com/jquery-1.10.1.js and it didn't work when I removed the code you've removed here.

Comment (Marsup) [inline deft.js:121]: @dtao Yeah I totally fucked up on this one :)
Please look at the new diff, it should all work.

**Comment by Marsup** [comment]
@dtao good to go ?

**Comment by Marsup** [comment]
@dtao sorry to bother you again, can we get this moving ?

**Pull request closed without merging**